src/TLuaInterpreter.cpp

```diff
@@ -15198,55 +15215,87 @@ void TLuaInterpreter::initIndenterGlobals()
+        "../src/mudlet-lua/lua/LuaGlobal.lua",
+        "mudlet-lua/lua/LuaGlobal.lua",
+        LUA_DEFAULT_PATH "/LuaGlobal.lua"
+    };
+    QStringList failedMessages{};
 
-    int error = luaL_dofile(pGlobalLua, path.toUtf8().constData());
-    if (error != 0) {
-        // For the installer we do not go down a level to search for this. So
-        // we check again for the user case of a windows install.
+    // load via Qt so UTF8 paths work on Windows - Lua can't handle it
+    auto readAllText = [=](const QString& path) -> QString {
+        QFile file(path);
+        if (!file.open(QFile::ReadOnly | QFile::Text)) {
+            qWarning() << "TLuaInterpreter::loadGlobal() ERROR: couldn't open " << path;
+            return QString();
+        }
 
-        // overload previous behaviour to check by absolute path as well
-        // TODO this sould be cleaned up and refactored to just use an array and a for loop
-        path = QCoreApplication::applicationDirPath() + "/mudlet-lua/lua/LuaGlobal.lua";
+        QTextStream in(&file);
+        QString text = in.readAll();
+        file.close();
+
+        return text;
+    };
+
+    int error;
+    for (const auto& path : qAsConst(pathsToTry)) {
         if (!QFileInfo::exists(path)) {
-            path = "mudlet-lua/lua/LuaGlobal.lua";
+            failedMessages << tr("%1 (doesn't exist)", "This file doesn't exist").arg(path);
+            continue;
         }
-        error = luaL_dofile(pGlobalLua, path.toUtf8().constData());
+
+        auto luaGlobal = readAllText(path);
+        if (luaGlobal.isEmpty()) {
+            failedMessages << tr("%1 (couldn't read file)", "This file could not be read for some reason (for example, no permission)").arg(path);
+            continue;
+        }
+
+        error = luaL_dostring(pGlobalLua, luaGlobal.toUtf8().constData());
         if (error == 0) {
             mpHost->postMessage(tr("[  OK  ]  - Mudlet-lua API & Geyser Layout manager loaded."));
             return;
+        } else {
+            qWarning() << "TLuaInterpreter::loadGlobal() loading " << path << " failed: " << lua_tostring(pGlobalLua, -1);
+            failedMessages << QStringLiteral("%1 (%2)").arg(path, lua_tostring(pGlobalLua, -1));
         }
-    } else {
-        mpHost->postMessage(tr("[  OK  ]  - Mudlet-lua API & Geyser Layout manager loaded."));
+    }
+
+    mpHost->postMessage(tr("[ ERROR ] - Couldn't to load LuaGlobal; your Mudlet is broken! Tried these locations:\n%1").arg(failedMessages.join(QStringLiteral("\n"))));
+}
+
+#if defined(Q_OS_WIN32)
+// No documentation available in wiki - internal function
+// loads utf8_filenames from the resource system directly so it is not affected by
+// non-ASCII characters that might be present in the users filesystem
+void TLuaInterpreter::loadUtf8Filenames()
+{
+    QFile file(QStringLiteral(":/mudlet-lua/lua/utf8_filenames.lua"));
+    if (!file.open(QFile::ReadOnly | QFile::Text)) {
```

Comment:
This seems to be the same as the `readAll` function above. Maybe make it a real function and use it here as well?

Reply:
I can, but I was unsure of where to put it. I'll keep it in `TLuaInterpreter` for now but it shouldn't be there...

Reply:
Not as `readAll` but maybe as `readScript`? I mean, duplicating that code isn't pretty either.